src/test/fuzz/http_request.cpp

```diff
@@ -17,8 +17,24 @@
 #include <string>
 #include <vector>
 
+// workaround for libevent versions before 2.1.1,
+// when internal functions didn't have underscores at the end
+#if _EVENT_NUMERIC_VERSION < 0x02010100
+extern "C" int evhttp_parse_firstline(struct evhttp_request*, struct evbuffer*);
+extern "C" int evhttp_parse_headers(struct evhttp_request*, struct evbuffer*);
+inline int evhttp_parse_firstline_(struct evhttp_request* r, struct evbuffer* b)
+{
+    return evhttp_parse_firstline(r, b);
+}
+inline int evhttp_parse_headers_(struct evhttp_request* r, struct evbuffer* b)
+{
+    return evhttp_parse_headers(r, b);
+}
+#else
 extern "C" int evhttp_parse_firstline_(struct evhttp_request*, struct evbuffer*);
```

Comment:
cc @practicalswift Can this fuzzer be adjusted to remove `evhttp_parse_firstline_`?

Reply:
Not that I know of but if someone can come up with a less "bizarre" solution that it would be awesome of course :)

Reply:
I think ideally, fuzzing libevent's internals belongs in libevent and not in our code. But I don't think it's a big problem to have it like this.

Reply:
@laanwj Just to clarify: the reason for the calls to the internal functions is not to fuzz libevent's internals but to be able to do `HTTPRequest http_request{evreq, true};` where the `evhttp_request` object is initialized in a way resembling the real-world attack surface. Let me know if you can think of a way to do that without calling internal functions - that would be great but I couldn't find a way to do it :)

---

Context:

```
    HTTPRequest http_request{evreq, true};
    const HTTPRequest::RequestMethod request_method = http_request.GetRequestMethod();
    (void)RequestMethodString(request_method);
    (void)http_request.GetURI();
    (void)http_request.GetHeader("Host");
```